src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -6466,14 +7143,18 @@ init_myhook(void)
 
 	register_a_hook("libc", "fcntl", (void *)new_fcntl, (long int *)(&libc_fcntl));
 
-	register_a_hook("libc", "mmap", (void *)new_mmap, (long int *)(&next_mmap));
-	register_a_hook("libc", "munmap", (void *)new_munmap, (long int *)(&next_munmap));
+	if (compatible_mode == false) {
+		register_a_hook("libc", "mmap", (void *)new_mmap, (long int *)(&next_mmap));
+		register_a_hook("libc", "munmap", (void *)new_munmap, (long int *)(&next_munmap));
+	}
 
 	register_a_hook("libc", "exit", (void *)new_exit, (long int *)(&next_exit));
 	register_a_hook("libc", "dup3", (void *)new_dup3, (long int *)(&libc_dup3));
 
 	init_fd_dup2_list();
 
+	extract_exe_path();
```

Comment:
As per this function it's only use is setting the `is_bash` flag and that's only used for compatible mode so to avoid extra overhead and complexity you should only set call this when the flag is enabled, assuming it's known at this point.

Reply:
Now check_exe_sh_bash() is called only in compatible mode. Thank you!